### ISSUE TITLE
Empty column after 41 players

### ISSUE THREAD

**Issue by bobhenl** [CodeCrafter47/BungeeTabListPlus#620]
Hi, somewhere after 41 players there's appearing empty column (40 players is ok):
![image](https://user-images.githubusercontent.com/18662086/107366176-1d318280-6ade-11eb-88b7-7a2b6dd11967.png)
https://paste.helpch.at/saduqakure.http Here's my file

Version: https://ci.codecrafter47.de/job/BungeeTabListPlus/1090/

**Comment by CodeCrafter47** [comment]
To avoid this, you need to configure the tab list such that all players on the server can be displayed.

**Comment by Andre601** [comment]
> To avoid this, you need to configure the tab list such that all players on the server can be displayed.

They do.
They use `currentserver` which is `${player server} == ${viewer server}` and therefore should list all players on the server (Unless you mean the network that is)

**Comment by CodeCrafter47** [comment]
That is correct. But the size is set to `40` slots, so no more that 40 players can be displayed with this config. You need to make that tab list larger to accommodate that many players.

**Comment by bobhenl** [comment]
@CodeCrafter47  and just question, why when I have 2 columns instead of 40 slots, why the 2 columns don't clip to the third one when there are fullfilled?
![image](https://user-images.githubusercontent.com/18662086/107533426-378f5d00-6bbf-11eb-815e-9fac746536fd.png)

https://paste.helpch.at/wupejudaqe.http

**Comment by bobhenl** [comment]
And just wanna report that even with new config (2 columns) instead of 40 players... it's still broken
![image](https://user-images.githubusercontent.com/18662086/107540022-e20a7e80-6bc5-11eb-9728-a9ddc0f8bd1d.png)

**Comment by CodeCrafter47** [comment]
Its the number of players on the spigot server, not the total number of players on the network that causes this.

**Comment by bobhenl** [comment]
Yeah, there were 41 players on spigot server, but this is bug, isn't it? This empty row....

And also when there's 8 more players on this server and it won't create third column, this is also bug, or?

**Comment by CodeCrafter47** [comment]
If I'm interpreting the numbers on your tab list correctly, in the picture with `8 more`, there are only 37 players on the spigot server, which fit into 40 slots.

It is debatable whether this is a bug. I choose not to see it as a bug, since it is nothing I can change. This is a limitation of minecraft. All players on the spigot server need to be displayed on the tab list. You configured your tab list to have only 40 slots. When there are more players the plugin adds a third column, and displays the heads of those players there. This behavior is very obvious in the very first image you shared.

I suggest you modify your config to use 60 or 80 slots, so that all your players fit on the tab list.

**Comment by bobhenl** [comment]
and where do I have 40 slots set? Do you mean columns: 2? Should I change it to columns 3? But then it will be always columns 3 instead of columns 2 that will change to columns 3 when full columns 2 would be reached.

**Comment by CodeCrafter47** [comment]
That is correct, it will always have 3 columns then. The alternative is to use the [dynamic size tab list type](https://github.com/CodeCrafter47/BungeeTabListPlus/wiki/Dynamic-Size-Tab-List), however it does not allow for custom slots.

**Comment by bobhenl** [comment]
mhm, and with 2 columns... shouldn't it show just for ex. ... and next XX players instead of third column? Or so it couldn't be anyhow solved than using more columns?

> And just wanna report that even with new config (2 columns) instead of 40 players... it's still broken
> ![image](https://user-images.githubusercontent.com/18662086/107540022-e20a7e80-6bc5-11eb-9728-a9ddc0f8bd1d.png)

**Comment by CodeCrafter47** [comment]
I'm afraid that is not possible.
This is a limitation of the Minecraft client itself, there is nothing a server side plugin such as BungeeTabListPlus can do about it.

**Comment by bobhenl** [comment]
So what do you suggest me? Increase column size or let it be? And could you still try to look if there isn't any solution? Maybe that when the specific player amount is on server that config could change (to increase columns), so it will show then 3 instead of 2 and one empty.

**Comment by CodeCrafter47** [comment]
I suggest you either increase the number of columns, or use the dynamic size tab list type. The information you currently display in the custom text slots at the top and bottom can be easily moved to the header and footer when using the dynamic size tab list type.

**Comment by Andre601** [comment]
What's the point of the column option then?
According to the wiki should it increase when more players are online than displayable on the server. Like this behaviour of the tab seems completely against what this option is meant to be: A hybrid between fixed size (custom slots) and dynamic size (Tab increasing)

**Comment by CodeCrafter47** [comment]
The point of the column option is that it allows the plugin to adjust the number of rows depending on how many players are online.

**Comment by bobhenl** [comment]
So this cannot be fixed anyway? :(
![image](https://user-images.githubusercontent.com/18662086/108540318-55587280-72e1-11eb-9d10-278f6b1cfbf9.png)


Except dynamic size

**Comment by CodeCrafter47** [comment]
Yes, I am afraid, that is the only option to fix it.
